crates/cxx-qt-build/src/lib.rs

```diff
@@ -536,110 +555,129 @@ impl CxxQtBuilder {
+            path,
+            moc_arguments,
+        } in &self.qobject_headers
+        {
+            let moc_products = qtbuild.moc(path, moc_arguments.clone());
+            self.cc_builder.file(moc_products.cpp);
         }
+    }
 
-        // Generate files
-        for files in generate_cxxqt_cpp_files(&self.rust_sources, &generated_header_dir) {
+    fn generate_cpp_files_from_cxxqt_bridges(&mut self, header_dir: impl AsRef<Path>) {
+        for files in generate_cxxqt_cpp_files(&self.rust_sources, &header_dir) {
             self.cc_builder.file(files.plain_cpp);
             if let (Some(qobject), Some(qobject_header)) = (files.qobject, files.qobject_header) {
                 self.cc_builder.file(&qobject);
                 self.qobject_headers.push(qobject_header.into());
             }
         }
+    }
 
-        // Run moc on C++ headers with Q_OBJECT macro
-        for QObjectHeaderOpts {
-            path,
-            moc_arguments,
-        } in self.qobject_headers
-        {
-            let moc_products = qtbuild.moc(&path, moc_arguments);
-            self.cc_builder.file(moc_products.cpp);
+    fn build_object_file(
+        builder: &cc::Build,
+        file_path: impl AsRef<Path>,
+        export_path: Option<(&str, &str)>,
+    ) {
+        let mut obj_builder = builder.clone();
+        obj_builder.file(file_path);
+        let obj_files = obj_builder.compile_intermediates();
+
+        if let [obj_file] = &obj_files[..] {
+            if let Some(export_dir) = export_dir() {
+                if let Some((out_directory, out_file_name)) = export_path {
+                    let obj_dir = export_dir.join(out_directory);
+                    std::fs::create_dir_all(&obj_dir).unwrap_or_else(|_| {
+                        panic!(
+                            "Could not create directory for object file: {}",
+                            obj_dir.to_string_lossy()
+                        )
+                    });
+                    let obj_path = obj_dir.join(out_file_name);
+                    std::fs::copy(obj_file, &obj_path).unwrap_or_else(|_| {
+                        panic!(
+                            "Failed to move object file to {obj_path}!",
+                            obj_path = obj_path.to_string_lossy()
+                        )
+                    });
+                }
+            } else {
+                println!("cargo::rustc-link-arg={}", obj_file.to_string_lossy());
+                // The linker argument order matters!
+                // We need to link the object file first, then link the static library.
+                // Otherwise, the linker will be unable to find the symbols in the static library file.
+                // See also: https://stackoverflow.com/questions/45135/why-does-the-order-in-which-libraries-are-linked-sometimes-cause-errors-in-gcc
+                println!("cargo::rustc-link-arg=-l{}", static_lib_name());
+            }
+        } else {
+            panic!(
```

Comment:
ah this is why you didn't use `first()`, a comment higher up saying we want to hit the else case if there is not one would be good.

Reply:
Done